gensim/matutils.py

```diff
@@ -699,13 +699,24 @@ def unitvec(vec, norm='l2'):
             return vec
 
     if isinstance(vec, np.ndarray):
```

Comment:
Please add tests for this function (checking that we normalized correctly & handle original data-type)